src/tests/networking_tests.rs

```diff
@@ -0,0 +1,1196 @@
+            let cage2 = {CAGE_TABLE.read().unwrap().get(&2).unwrap().clone()};
+            let mut socket2 = interface::GenSockaddr::V4(interface::SockaddrV4::default());
+            assert!(cage2.accept_syscall(serversockfd, &mut socket2) > 0); //really can only make sure that the fd is valid
+            
+            assert_eq!(cage2.close_syscall(serversockfd), 0);
+            assert_eq!(cage2.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
+        });
+
+        interface::sleep(interface::RustDuration::from_millis(100));
+        assert_eq!(cage.connect_syscall(clientsockfd, &socket), 0); 
+        
+        let mut retsocket = interface::GenSockaddr::V4(interface::SockaddrV4::default());
+        assert_eq!(cage.getsockname_syscall(clientsockfd, &mut retsocket), 0);
+        assert_ne!(retsocket, socket);
+
+        assert_eq!(cage.close_syscall(serversockfd), 0);
+        
+        thread.join().unwrap();
+        
+        assert_eq!(cage.exit_syscall(EXIT_SUCCESS), EXIT_SUCCESS);
+        lindrustfinalize();
+    }
+
+
+
+    pub fn ut_lind_net_poll() {
+        lindrustinit();
+        let cage = {CAGE_TABLE.read().unwrap().get(&1).unwrap().clone()};
+
+        let filefd = cage.open_syscall("/netpolltest.txt", O_CREAT | O_EXCL | O_RDWR, S_IRWXA);
+        assert!(filefd > 0);
+
+        let serversockfd = cage.socket_syscall(AF_INET, SOCK_STREAM, 0);
+        let clientsockfd1 = cage.socket_syscall(AF_INET, SOCK_STREAM, 0);
+        let clientsockfd2 = cage.socket_syscall(AF_INET, SOCK_STREAM, 0);
+
+        let port: u16 = 53009;
+        let sockaddr = interface::SockaddrV4{ sin_family: AF_INET as u16, sin_port: port.to_be(), sin_addr: interface::V4Addr{ s_addr: u32::from_ne_bytes([127, 0, 0, 1]) }, padding: 0};
+        let socket = interface::GenSockaddr::V4(sockaddr); //127.0.0.1 from bytes above
+        assert_eq!(cage.bind_syscall(serversockfd, &socket), 0);
+        assert_eq!(cage.listen_syscall(serversockfd, 4), 0); 
+
+        let serverpoll = interface::PollStruct{fd: serversockfd, events: POLLIN, revents: 0};
+        let filepoll = interface::PollStruct{fd: filefd, events: POLLIN, revents: 0};
+        let mut polled = vec![serverpoll, filepoll];
+
+        cage.fork_syscall(2);
+        //client 1 connects to the server to send and recv data...
+        let thread1 = interface::helper_thread(move || {
+            interface::sleep(interface::RustDuration::from_millis(100));
+            let cage2 = {CAGE_TABLE.read().unwrap().get(&2).unwrap().clone()};
+
+            assert_eq!(cage2.connect_syscall(clientsockfd1, &socket), 0);
+            assert_eq!(cage2.send_syscall(clientsockfd1, str2cbuf(&"test"), 4, 0), 4);
+            //giving it a longer pause time to that it can process all of the data that it is recieving
+            interface::sleep(interface::RustDuration::from_millis(1000));
+
+            assert_eq!(cage2.close_syscall(serversockfd), 0);
+            cage2.exit_syscall();
```

Comment:
These will fail, since exit_syscall expects a status variable.
`cage2.exit_syscall(EXIT_SUCCESS)`

Reply:
This issue appears several times in networking_tests, a search for `exit_syscall()` should bring all the occurrences.